pennylane_lightning_gpu/src/simulator/interfaces/torch/SVCudaTorch.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+#include <cuComplex.h>
+#include <torch/extension.h>
+
+#include "StateVectorCudaRaw.hpp"
+
+/// @cond DEV
+namespace {
+namespace cuUtil = Pennylane::CUDA::Util;
+using namespace Pennylane::CUDA;
+using namespace Pennylane::Util;
+} // namespace
+/// @endcond
+
+namespace Pennylane {
+
+/**
+ * @brief Raw memory CUDA state-vector class using custateVec backed
+ * gate-calls.
+ *
+ * @tparam Precision Floating-point precision type.
+ */
+template <class Precision>
+class SVCudaTorch final : public StateVectorCudaRaw<Precision> {
```

Comment:
I am curious what is the exact difference between this class and `StateVectorCudaRaw`. Can we just use `StateVectorCudaRaw` by providing a function converting `torch::tensor` to `StateVectorCudaRaw`?

Reply:
Aha I see. So we anyway need to keep `torch::tensor` instance for a lifetime.